src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -19,6 +19,22 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
   /** The name of the phase: */
   val phaseName: String = "mixin"
 
+  /** Some trait methods need to be implemented in subclasses, so they cannot be private.
+    *
+    * They may be protected, now that traits are compiled 1:1 to interfaces.
+    *
+    * TODO: interfaces can also have private members, so there's also less need to make trait members non-private
+    * can we leave more methods private?
+    * (they still may need to be implemented in subclasses, though we could make those protected...).
```

Comment:
> though we could make those protected

Sadly, [not](https://docs.oracle.com/javase/specs/jvms/se8/jvms8.pdf):

> Methods of interfaces may have any of the flags in Table 4.6-A set except
ACC_PROTECTED, ACC_FINAL, ACC_SYNCHRONIZED, and ACC_NATIVE (JLS §9.4).